rustls/src/builder.rs

```diff
@@ -210,6 +211,38 @@ impl<S: ConfigSide> ConfigBuilder<S, WantsVersions> {
             return Err(Error::General("no kx groups configured".into()));
         }
 
+        // verifying cipher suites have matching kx groups
+        let mut supported_kx_algos = Vec::with_capacity(ALL_KEY_EXCHANGE_ALGORITHMS.len());
```

Comment:
Hmm, reviewing this here, IIRC we want to avoid referencing `ALL_` constants in this way, because we should avoid linking algorithms that aren't used. Maybe use `provider.kx_groups.len()` instead and drop the optimization?

Reply:
I think `ALL_KEY_EXCHANGE_ALGORITHMS` doesn't link in the actual algorithms, it is just a static list of `KeyExchangeAlgorithm` enums.

Reply:
Can this feedback thread be resolved then? It seems like yes